cluster-operator/src/test/java/io/strimzi/operator/cluster/model/CruiseControlTest.java

```diff
@@ -0,0 +1,578 @@
+                .withImage(ccImage)
+                .withNewTemplate()
+                    .withNewCruiseControlContainer()
+                        .withEnv(envVar1, envVar2)
+                    .endCruiseControlContainer()
+                .endTemplate()
+                .build();
+
+        Kafka resource = ResourceUtils.createKafkaCluster(namespace, cluster, replicas, image,
+                healthDelay, healthTimeout, metricsCm, kafkaConfig, zooConfig,
+                kafkaStorage, zkStorage, null, kafkaLogJson, zooLogJson, null, cruiseControlSpec);
+        CruiseControl cc = CruiseControl.fromCrd(resource, VERSIONS);
+
+        List<EnvVar> kafkaEnvVars = cc.getEnvVars();
+        assertThat(kafkaEnvVars.stream().filter(var -> testEnvOneKey.equals(var.getName())).map(EnvVar::getValue).findFirst().get(), is(testEnvOneValue));
+    }
+
+    @Test
+    public void testCruiseControlNotDeployed() {
+        Kafka resource = ResourceUtils.createKafkaCluster(namespace, cluster, replicas, image,
+                healthDelay, healthTimeout, metricsCm, kafkaConfig, zooConfig,
+                kafkaStorage, zkStorage, null, kafkaLogJson, zooLogJson, null,  null);
+        CruiseControl cc = CruiseControl.fromCrd(resource, VERSIONS);
+
+        try {
+            assertThat(cc.generateDeployment(true, null, null, null), is(nullValue()));
+            assertThat(cc.generateService(), is(nullValue()));
+            assertThat(cc.generateSecret(null, true), is(nullValue()));
+        } catch (Throwable expected) {
+            assertEquals(NullPointerException.class, expected.getClass());
+        }
+    }
+
+    @Test
+    public void testGenerateService()   {
+        Service svc = cc.generateService();
+
+        assertThat(svc.getSpec().getType(), is("ClusterIP"));
+        assertThat(svc.getMetadata().getLabels(), is(expectedLabels(cc.getServiceName())));
+        assertThat(svc.getSpec().getSelector(), is(expectedSelectorLabels()));
+        assertThat(svc.getSpec().getPorts().size(), is(1));
+        assertThat(svc.getSpec().getPorts().get(0).getName(), is(CruiseControl.REST_API_PORT_NAME));
+        assertThat(svc.getSpec().getPorts().get(0).getPort(), is(new Integer(CruiseControl.REST_API_PORT)));
+        assertThat(svc.getSpec().getPorts().get(0).getProtocol(), is("TCP"));
+
+        checkOwnerReference(cc.createOwnerReference(), svc);
+    }
+
+    @Test
+    public void testGenerateServiceWhenDisabled()   {
+        Kafka resource = ResourceUtils.createKafkaCluster(namespace, cluster, replicas, image,
+                healthDelay, healthTimeout, metricsCm, kafkaConfig, zooConfig,
+                kafkaStorage, zkStorage, null, kafkaLogJson, zooLogJson, null, null);
+        CruiseControl cc = CruiseControl.fromCrd(resource, VERSIONS);
+
+        try {
+            assertThat(cc.generateService(), is(nullValue()));
+        } catch (Throwable expected) {
+            assertEquals(NullPointerException.class, expected.getClass());
```

Comment:
Just to be a little more consistent with the rest of the tests in Strimzi would you opposed to using
```
assertThrows(NullPointerException.class, () -> cc.generateService());
```